Tests/BackendIntegrationTests/BaseBackendIntegrationTests.swift

```diff
@@ -156,7 +172,14 @@ private extension BaseBackendIntegrationTests {
 
     private var dangerousSettings: DangerousSettings {
         return .init(autoSyncPurchases: true,
-                     internalSettings: .init(enableReceiptFetchRetry: true))
+                     internalSettings: self)
     }
 
 }
+
+extension BaseBackendIntegrationTests: InternalDangerousSettingsType {
+
+    var enableReceiptFetchRetry: Bool { return true }
+    var forceServerErrors: Bool { return false }
```

Comment:
I think composition over inheritance is the way to go here.

It feels unnatural for the test class itself, which is fairly large and complex, to conform to this protocol and being what gets "dependency-injected" into the `DangerousSettings` initializer.

Why not have `BaseBackendIntegrationTests` instances hold a member variable of a new type, say a `MutableInternalDangerousSettings` class, that conforms to this protocol? 

The `serverUp()` and `serverDown()` methods can then just mutate `forceServerErrors` that variable.

Reply:
> I think composition over inheritance is the way to go

Normally I agree, but I fail to see how that would improve this particular case. That would semantically be the same.
This set up is much simpler this way. The way these tests are set up is such that the tests themselves become the configuration, which allows us to override flags like [this](https://github.com/RevenueCat/purchases-ios/blob/662940100f0cd1f4005934d2a99b4ba16e9ea6d3/Tests/BackendIntegrationTests/StoreKitObserverModeIntegrationTests.swift#L25).

What benefit does your suggestion provide?